addons/account/models/account_move.py

```diff
@@ -5483,13 +5518,27 @@ def is_line_reconciled(line):
                         results['exchange_partials'] += exchange_diff_partials
 
             # ==== Create the full reconcile ====
-
             results['full_reconcile'] = self.env['account.full.reconcile'].create({
                 'exchange_move_id': exchange_move and exchange_move.id,
                 'partial_reconcile_ids': [(6, 0, involved_partials.ids)],
                 'reconciled_line_ids': [(6, 0, involved_lines.ids)],
             })
 
+            # === Cash basis rounding autoreconciliation ===
+            # In case a cash basis rounding difference line got created for the transition account, we reconcile it with the corresponding lines
+            # on the cash basis moves (so that it reaches full reconciliation and creates an exchange difference entry for this account as well)
+
+            if caba_lines_to_reconcile:
+                for (move, account, repartition_line), amls_to_reconcile in caba_lines_to_reconcile.items():
+                    if not account.reconcile:
+                        continue
+
+                    exchange_line = exchange_move.line_ids\
+                                    .filtered(lambda l: l.account_id == account and l.tax_repartition_line_id == repartition_line) if exchange_move \
+                                    else self.env['account.move.line']
```

Comment:
```py
if exchange_move:
    exchange_line = exchange_move.line_ids\
        .filtered(lambda l: l.account_id == account and l.tax_repartition_line_id == repartition_line)
else:
    exchange_line = self.env['account.move.line']
```
maybe?

Reply:
The `if ... else` is useless actually
```suggestion
                    exchange_line = exchange_move.line_ids\
                                    .filtered(lambda l: l.account_id == account and l.tax_repartition_line_id == repartition_line)
```